### ISSUE TITLE
Sigma Weights Tensor not available after first batch

### ISSUE THREAD

**Issue by DHOFM** [Derposoft/crypto_pump_and_dump_with_deep_learning#12]
Hi,
first I would like to say thanks a lot for your great anomaly implementation of https://arxiv.org/pdf/2110.02642.pdf

Every Model works fine for me, except of the anomaly Transformer - the AnomalyTransfomerIntermediate works also.
Training the AnomalyTransformer, the first batch iteration works. After that, the sigma gets lost. I debugged into PyTorch Forward Function and there are no more weights. This is strange behavior, I tested locally with GPU and with colab. 
Here is a screenshot of the debugger:

![image](https://user-images.githubusercontent.com/27775323/200866060-40c994aa-fc1c-48d9-85a7-32b441a9e65e.png)

This is called (as you know) by:  self.sigma = self.Ws(x)

I am more into Tensorflow than into PyTorch but anyway I cannot find the problem in your code. To me it looks correct (Attention, Transformer Blocks and so on)

Kind regards,
Dirk

**Comment by gireesh-m** [comment]
Ack

**Comment by gireesh-m** [comment]
So it looks like the this originates from the way the loss is computed in the full anomaly transformer. Since the authors of the original paper have made some tweaks to the model, this is probably a good time to just rewrite the whole AT. Will start working on it tomorrow.

**Comment by Derposoft** [comment]
@DHOFM Sorry for the delay on this issue. I know this is an old thread and I imagine you aren't working with this anymore, but if you are, I just pushed a change to the AT models that should fix this. Let us know if you still find issues with it -- otherwise, I'm planning on closing this issue in a week or so. Thanks for bringing this up!
